components/website/utils/Schedule/Table/index.jsx

```diff
@@ -1,40 +1,68 @@
 import Block from './Block';
 import { isSelected } from '../Day/Filters'
-import { useState } from 'react'
 
 function filterElem(filters)
 {
     return function(elem)
     {
-        if (filters === "")
+        if (filters == "")
+            return true;
+
+        let result = elem.activity.author == filters;
+        if(result)
             return true;
 
-        //TODO
         switch (elem.activity.activityType)
-        {
-            case "Coffee Break":    return isSelected(filters, "Breaks");
-            case "Talk":            return isSelected(filters, "Talks");
-            default:                return false;
+        {   
+            case "Coffee Break":    result = isSelected(filters, "Breaks"); break;
+            case "Talk":            result = isSelected(filters, "Talks"); break;
+            case "Pitch":           result = isSelected(filters, "Pitch"); break;
+            case "Workshop":        result = isSelected(filters, "Workshops"); break;
+            case "Hackathon":       result = isSelected(filters, "Hackathons"); break;
+            default:                break;
         }
+        return result;
     }
 }
 
-export default function Table(props)
-{
+/*
+ *  Groups the activities into arrays. Two elements will be in the same array if they happen at the same
+ *  time
+ */
+function group(list) {
+    var result = [];
```

Comment:
```suggestion
    const result = [];
```

From the code, I think your result will be a const variable, only its contents are going to change.